var/spack/repos/builtin/packages/r-ergm/package.py

```diff
@@ -12,16 +12,22 @@ class RErgm(RPackage):
        part of the "statnet" suite of packages for network analysis."""
 
     homepage = "http://statnet.org"
-    url      = "https://cran.r-project.org/src/contrib/ergm_3.7.1.tar.gz"
-    list_url = "https://cran.r-project.org/src/contrib/Archive/ergm"
+    url      = "https://cloud.r-project.org/src/contrib/ergm_3.7.1.tar.gz"
+    list_url = "https://cloud.r-project.org/src/contrib/Archive/ergm"
 
+    version('3.10.4', sha256='885f0b1a23c5a2c1947962350cfab66683dfdfd1db173c115e90396d00831f22')
+    version('3.10.1', sha256='a2ac249ff07ba55b3359242f20389a892543b4fff5956d74143d2d41fa6d4beb')
     version('3.7.1', '431ae430c76b2408988f469831d80126')
 
-    depends_on('r-robustbase@0.9-10:', type=('build', 'run'))
-    depends_on('r-coda@0.18-1:', type=('build', 'run'))
-    depends_on('r-trust', type=('build', 'run'))
-    depends_on('r-matrix', type=('build', 'run'))
-    depends_on('r-lpsolve', type=('build', 'run'))
-    depends_on('r-mass', type=('build', 'run'))
+    depends_on('r-robustbase@0.93-5:', type=('build', 'run'))
+    depends_on('r-coda@0.19-2:', type=('build', 'run'))
+    depends_on('r-trust@01-7:', type=('build', 'run'))
+    depends_on('r-matrix@1.2-17:', type=('build', 'run'))
+    depends_on('r-lpsolve@5.6.13:', type=('build', 'run'))
+    depends_on('r-mass@7.3-51.4:', type=('build', 'run'))
     depends_on('r-statnet-common@3.3:', type=('build', 'run'))
-    depends_on('r-network@1.13:', type=('build', 'run'))
+    depends_on('r-network@1.15:', type=('build', 'run'))
+    depends_on('r-purrr@0.3.2:', when='@3.10.0:', type=('build', 'run'))
+    depends_on('r-rlang@0.3.4:', when='@3.10.0:', type=('build', 'run'))
+    depends_on('r-tibble@2.1.1:', when='@3.10.0:', type=('build', 'run'))
+    depends_on('r-dplyr@0.8.0.1:', when='@3.10.0:', type=('build', 'run'))
```

Comment:
Missing a system dependency:

> SystemRequirements:	OpenMPI

You can probably use any generic MPI

Reply:
@adamjstewart This is an interesting one and I would appreciate your input. The OpenMPI system requirement was added for version 3.10.4, the latest version. However, looking at the documentation, and the source, it is not really a dependency but more of, if it detects that it is running under MPI, ie., started with mpirun, then it will use it. However, in order to use MPI it would also need to have the Rmpi (r-rmpi) and the doSNOW (r-dosnow) packages, and by extension, the snow (r-snow) packages installed. None of those actually list MPI as a dependency for installation of the respective R packages. Furthermore, those are not listed as dependencies for r-ergm, although Rmpi is in the suggested list.

If one follows the MPI example in the docs it is clear what packages one would need to install and load to use MPI, and nothing in r-ergm has to be built beforehand for that support, so it does seem to be unnecessary from a packaging perspective. I suppose that a variant could be used to ensure that the MPI related R packages can be optionally installed with r-ergm, but the two variants would be the exact same package, just one installing extra packages along the way. I lean towards leaving out the MPI dependency but I am not sure if there have been other cases like this in Spack that set precedent.

Reply:
Based on a related comment about the r-snow package, it seems like leaving the OpenMPI (MPI) dependency out is the best course to take.

Reply:
I added a comment about using MPI with this package.

Reply:
I agree with this decision.